ducktape/command_line/main.py

```diff
@@ -35,9 +39,25 @@ def parse_args():
                         help='one or more space-delimited strings indicating where to search for tests')
     parser.add_argument("--collect-only", action="store_true", help="display collected tests, but do not run")
     parser.add_argument("--debug", action="store_true", help="pipe more verbose test output to stdout")
+    parser.add_argument("--config-file", action="store", default=ConsoleConfig.PROJECT_CONFIG_FILE,
+                        help="path to project-specific configuration file")
+    parser.add_argument("--cluster", action="store", default=ConsoleConfig.CLUSTER_TYPE,
+                        help="cluster class to use to allocate nodes for tests")
     parser.add_argument("--exit-first", action="store_true", help="exit after first failure")
 
-    args = parser.parse_args()
+    project_configs = []
+    project_config_file = ConsoleConfig.PROJECT_CONFIG_FILE
+    if os.path.exists(project_config_file):
+        project_configs = list(itertools.chain(*[line.split() for line in open(project_config_file).readlines()]))
+
+    user_configs = []
+    user_config_file = os.path.expanduser(ConsoleConfig.USER_CONFIG_FILE)
+    if os.path.exists(user_config_file):
+        user_configs = list(itertools.chain(*[line.split() for line in open(user_config_file).readlines()]))
+
+    # The order of concatenation ensures overrides are handled properly -- for action=store arguments, the later options
+    # will override earlier instances
+    args = parser.parse_args(project_configs + user_configs + sys.argv[1:])
```

Comment:
Oops, I read this as disagreeing with the comment but I was mistaken. 

This is definitely unintuitive to me however. I guess my mental model is greater specificity == higher priority. I would expect to set up my user config for common cross-project configs and override configs as needed at the project level. 

For example, if user config overrides project config and I want some config to be *different* between projects I have to: a) check whether there is a user config file b) delete relevant config there c) add that deleted config back to all the other projects d) update the one config on the one project that needs to be different.

Maybe it's moot in that it boils down to preference as to which types of config changes you prefer being less convenient. 

That said, however overrides work, the resulting "composite config" should be immediately visible (maybe print this in main?). This helps avoid cases where you think you're running with some configuration but it turns out to be different due to misunderstanding of override behavior, an unexpected config file in an unexpected place, or say a forgotten --no-teardown that was intended to be deleted but wasn't.